### PR TITLE
task-driver: integration: settle-match: Add settle-match-internal tests

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds tests for the `settle-match-internal` task in the task driver. This task is responsible for settling a match generated by the internal matching engine and updating the global state to reflect this update.

Going to add more tests in a follow up, this PR mostly introduces useful scaffolding.

### Testing
- Unit and integration tests pass
- Tested a simple settle match process

### CODE REVIEW

File: task-driver/integration/tests/settle_match.rs [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+fn dummy_order(side: OrderSide, test_args: &IntegrationTestArgs) -> Order {
+    let worst_cast_price = match side {
+        OrderSide::Buy => 12,
+        OrderSide::Sell => 9,
+    };
+
+    Order {
+        quote_mint: biguint_from_hex_string(&test_args.erc20_addr),
+        base_mint: biguint_from_hex_string(WETH_ADDR),
+        side: OrderSide::Buy,
```
Comment (akirillo) [inline task-driver/integration/tests/settle_match.rs:56]: shouldn't this be the `side` arg that was passed in

File: task-driver/integration/tests/settle_match.rs [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    let price = FixedPoint::from_f64_round_down(EXECUTION_PRICE);
+    let base_amount = 10;
+    let quote_amount = price * Scalar::from(base_amount);
+
+    let match_ = MatchResult {
+        quote_mint: biguint_from_hex_string(&test_args.erc20_addr),
+        base_mint: biguint_from_hex_string(WETH_ADDR),
+        quote_amount: scalar_to_u64(&quote_amount.floor()),
+        base_amount,
+        direction: 0,
```
Comment (akirillo) [inline task-driver/integration/tests/settle_match.rs:171]: perhaps worth documenting that this method expects party 0 in the match to be buy-side?

**Review by akirillo — APPROVED** [review]
one change reqd, one small doc nit, o/w lgtm

File: task-driver/integration/tests/settle_match.rs (hunk 2 above)
Comment (joeykraut) [reply task-driver/integration/tests/settle_match.rs]: changed it to be based on the arguments passed in

**Pull request merged**